libkbfs/backpressure_disk_limiter.go

```diff
@@ -470,6 +479,49 @@ func (bdl *backpressureDiskLimiter) onBlocksDelete(
 	bdl.fileTracker.onBlocksDelete(blockFiles)
 }
 
+func (bdl *backpressureDiskLimiter) onDiskBlockCacheDelete(
+	ctx context.Context, blockBytes int64) {
+	if blockBytes == 0 {
+		return
+	}
+	bdl.lock.Lock()
+	defer bdl.lock.Unlock()
+	bdl.byteTracker.onBlocksDelete(blockBytes)
+}
+
+func (bdl *backpressureDiskLimiter) beforeDiskBlockCachePut(
+	ctx context.Context, blockBytes, diskBlockCacheBytes int64) (
+	availableBytes int64, err error) {
+	if blockBytes == 0 {
+		// Better to return an error than to panic in ForceAcquire.
+		return 0, errors.New("backpressureDiskLimiter.beforeDiskBlockCachePut" +
+			" called with 0 blockBytes")
+	}
+	bdl.lock.Lock()
+	defer bdl.lock.Unlock()
+	freeBytes, freeFiles, err := bdl.freeBytesAndFilesFn()
+	if err != nil {
+		return 0, err
+	}
+	bdl.byteTracker.updateFree(freeBytes)
+	bdl.fileTracker.updateFree(freeFiles)
+
+	diskBlockCacheLimit := int64(bdl.byteTracker.currLimit() *
+		bdl.byteTracker.minThreshold * diskBlockCacheFrac)
+	if diskBlockCacheLimit < diskBlockCacheBytes+blockBytes {
+		// The overall disk block cache limit has been exceeded.  Return a
+		// negative result so the block cache knows how much it must evict
+		// before retrying.
+		return diskBlockCacheLimit - (diskBlockCacheBytes + blockBytes),
+			errors.New("backpressureDiskLimiter needs more space")
+	}
+	// The disk block cache is using up less than diskBlockCacheLimit bytes, so
+	// we can add more, even if it makes backpressure worse. This can return a
+	// negative number with a nil error, in which case the disk block cache
+	// must Put the block. It can choose to evict anyway afterwards.
+	return bdl.byteTracker.beforeDiskBlockCachePut(ctx, blockBytes)
```

Comment:
Consider that the common use case for this scenario is blocks being flushed from the journal, thus being `Put` into the disk cache. It should be a zero-sum operation (within reason).